### PR TITLE
feat: add pagination to getWorkspaces

### PR DESCRIPTION
**Pull request by f0ssel**
Backend for workspace page - part of https://github.com/coder/coder/issues/3907

### CODE REVIEW

File: coderd/workspaces.go [hunk 1]
```diff
@@ -1072,11 +1077,18 @@ func validWorkspaceSchedule(s *string, min time.Duration) (sql.NullString, error
 
 // workspaceSearchQuery takes a query string and returns the workspace filter.
 // It also can return the list of validation errors to return to the api.
-func workspaceSearchQuery(query string) (database.GetWorkspacesParams, []codersdk.ValidationError) {
+func workspaceSearchQuery(query string, page codersdk.Pagination) (database.GetWorkspacesParams, []codersdk.ValidationError) {
+	if page.Limit < 1 {
+		page.Limit = 65536
```
Comment (deansheather) [inline coderd/workspaces.go:1082]: why so big?

**Review by deansheather — APPROVED** [review]

File: coderd/workspaces.go (hunk 1 above)
Comment (f0ssel) [reply coderd/workspaces.go:1082]: I can't have it be a negative number, or 0, but I effectively don't want it to apply. This was 2^16 which seemed high enough to not practically get in the way.

Comment (sreya) [reply coderd/workspaces.go:1082]: if the limit is 0 why not just omit it from the query?

Comment (f0ssel) [reply coderd/workspaces.go:1082]: I wasn't sure how to do that exactly with the limit clause in postgres / sqlc - if you know of a way please share

Comment (sreya) [reply coderd/workspaces.go:1082]: You can use a case statement
```
select 
	*
from
	organizations
limit case when @limit > 0 then @limit end;
```

**Review by Emyrk — COMMENTED** [review]
~~Can we add this to `GetAuthorizedWorkspaces` too?~~

Nice :+1:

**Review by presleyp — APPROVED** [review]

**Pull request merged**